### PR TITLE
Refactoring Test Suite async utility functions - Part I

### PR DESCRIPTION
**Pull request by bh2smith**
This PR introduces a few useful async functions. 

- `makeDeposits`
- `placeOrders`
- `setupGenericStableX`

With these three functions alone, we have already cut 1000 lines of redundancy out of our test file.

### CODE REVIEW

File: test/resources/stablex_utils.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  return instance
+}
+
+
+/**
+ * Makes deposit transactions from a list of Deposit Objects
+ * @param {contract} - StablecoinConverter Smart Contract
+ * @param {accounts} - an array of (unlocked) ethereum account addresses
+ * @param {Deposit[]} - an array of Deposit Objects
+ */
```
Comment (nlordell) [inline test/resources/stablex_utils.js]: I think here the types aren't what I expected. And the parameter names are missing. Maybe something like:
```
 * @param {object} contract StablecoinConverter smart contract
 * @param {string[]} accounts An array of ...
 * @param {Deposit[]} depolitList An array of ...
```

From what I understand, JSDoc format is `@param {type} name description`.

File: test/resources/stablex_utils.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    await contract.deposit(tokenAddress, deposit.amount, { from: accounts[deposit.user] })
+  }
+}
+
+/**
+ * Makes placeOrder transactions from a list of Order Objects
+ * @param {contract} - StablecoinConverter Smart Contract
+ * @param {accounts} - an array of (unlocked) ethereum account addresses
+ * @param {Order[]} - an array of Order Objects
+ * @returns {BN[]}
```
Comment (nlordell) [inline test/resources/stablex_utils.js]: Same as above

**Comment by nlordell** [comment]
Looks great so far!

**Review by josojo — APPROVED** [review]
What  a clean up! great.

Reviewing some tests was not easy, as they were first deleted in one place and then inserted somewhere else, but since coverage is at 100%, you did not forget something, I guess.

Great!

**Comment by fleupold** [comment]
This is a good improvement! 

Generally, I'd be careful about optimizing the test cases a lot for "don't repeat yourself". It can easily lead to very complex to understand test cases (calling into many functions, with many different params/flavours to accomodate all test cases). Just reading a rather long test case gives you all the information you need in one go.

But here you are right, that our test file is so large and the parts that happen all the time should be made common.

**Pull request merged**